hypothesis-python/src/hypothesis/extra/pandas/impl.py

```diff
@@ -175,13 +177,17 @@ def range_indexes(
     * min_size is the smallest number of elements the index can have.
     * max_size is the largest number of elements the index can have. If None
       it will default to some suitable value based on min_size.
+    * name is the name of the index. Unlike the other index names, this should
+        be a string rather than a search strategy. If None, the index will have no name.
```

Comment:
Strong preference to exactly match the API of `indexes()` here, by accepting a strategy instead of a string.  That does mean you'd need to wrap `st.just()` around a constant string name, but it's the way we've gone here and much easier than the reverse need for `.flatmap()` or similar.

Reply:
I tried going down this route initially but found it a bit tricky - would you then advise to create a composite strategy of some kind to then call `.map()` on it to get the RangeIndex object? I tried this initially with the `pands.RangeIndex.from_range` class method, as `.map()` only works with positional arguments

Reply:
`st.builds(pandas.RangeIndex, st.integers(min_size, max_size), name=name)` should do it!  

And `check_strategy(name)` for validation; with a failing test for passing a bare string or None in `pandas/test_argument_validation.py`